model/src/presignature.rs

```diff
@@ -0,0 +1,45 @@
+//! Contains the Presignature type as described by the specification with serialization as described by the openapi documentation.
+
+use crate::order::OrderUid;
+use primitive_types::H160;
+use serde::{Deserialize, Serialize};
+
+#[derive(Eq, PartialEq, Clone, Debug, Default, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct PreSignature {
+    pub block_number: u64,
+    pub log_index: u64,
+    pub owner: H160,
+    pub order_uid: OrderUid,
+    pub signed: bool,
+}
```

Comment:
Maybe I don't understand the purpose of Presignature, but it is my impression that this will be used for smart contract wallets such as a multisig. With this in mind, I was under the impression that we would want to collect multiple signatures. Does this mean that this table is a many-one relation ship with the orders table? Or does each signature here correspond to a unique order in the orders table that can be combined together to make a single singed order?

Reply:
It is many to one (for every order there might be multiple "signatures"; however for every signature there is only one order). In practice the database query in `presignatures()` is built so that also for every order (uid) there is at most one "real" signature for an order (older "signatures" in the table are ignored). So if this table is only read using the function introduced in this PR, then the relation is effectively one to one.

Reply:
In that case my question would be that I don't see any signature being stored here. I can only guess then that when new signatures come in that the order is updated with the accumulated signatures so that, once the threshold is met, the order eventually becomes executable? 

If that is the case, do we have some threshold verification logic coming in later that will classify these orders as executable?

Reply:
I think the name `presignatures` is confusing for the table, I'll rename it to `presignature_events` to make it clear that no actual signature is stored there.
There is no "threshold" to speak of: an order is signed if the user has sent a `setPreSignature` call to the contract with `true` and the order uid as the input values. Any future call with the same `orderUid` overwrites the previous one. Validity of the order will have to be rechecked every time that the order is used again, as a previously valid order might have become invalid. All this logic will come later.